src/Database/Type.php

```diff
@@ -21,7 +21,7 @@
  * Encapsulates all conversion functions for values coming from database into PHP and
  * going from PHP into database.
  */
-class Type
+class Type implements TypeInterface
```

Comment:
I think the long term idea here is the mentioned above "only use it as the type builder" thing. So the base class will get more restricted soon then.